src/components/DeeplinkWrapper/index.website.js

```diff
@@ -21,68 +21,76 @@ const propTypes = {
 
         /** Currently logged-in user authToken */
         authToken: PropTypes.string,
+
+        /** The short-lived auth token for navigating to desktop app */
+        shortLivedAuthToken: PropTypes.string,
     }),
 };
 
 const defaultProps = {
     session: {
         email: '',
         authToken: '',
+        shortLivedAuthToken: '',
     },
 };
 
 class DeeplinkWrapper extends PureComponent {
     constructor(props) {
         super(props);
-
-        this.state = {
-            appInstallationCheckStatus: (this.isMacOSWeb() && CONFIG.ENVIRONMENT !== CONST.ENVIRONMENT.DEV)
-                ? CONST.DESKTOP_DEEPLINK_APP_STATE.CHECKING : CONST.DESKTOP_DEEPLINK_APP_STATE.NOT_INSTALLED,
-        };
-        this.focused = true;
+        this.hasPopupBeenOpenedBefore = !this.isMacOSWeb() || CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV;
     }
 
     componentDidMount() {
-        if (!this.isMacOSWeb() || CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV) {
+        if (this.hasPopupBeenOpenedBefore) {
             return;
         }
 
-        window.addEventListener('blur', () => {
-            this.focused = false;
-        });
+        // Since there is no way to know if a previous short-lived authToken is still valid,
+        // any previous short-lived authToken must be cleared out and a new one must be fetched
+        // so that the popup window will only open when we know the short-lived authToken is valid.
+        Session.removeShortLivedAuthToken();
 
-        const expensifyUrl = new URL(CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL);
-        const params = new URLSearchParams();
-        params.set('exitTo', `${window.location.pathname}${window.location.search}${window.location.hash}`);
+        // If the current link is transition from oldDot, we should wait for it to finish,
+        // because we should pass the final path to the desktop app instead of the transition path.
```

Comment:
Can we update this comment to explain what it means for us to "wait for it to finish"? It isn't obvious to me what that means or why we need to do that.

>we should pass the final path to the desktop app instead of the transition path

I'm pretty confused by this statement too. I don't see anything getting 'passed' here.